frontend/src/profile/Profile.tsx

```diff
@@ -0,0 +1,440 @@
+import { TextField } from "@material-ui/core";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import { ChangePasswordModal } from "./ChangePasswordModal"
+import { IconButton } from '@material-ui/core';
+import EditIcon from '@material-ui/icons/Edit';
+import {
+    setMajorAction,
+    setFullNameAction,
+    setEmailAction,
+    setUserCoopCycleAction
+} from "../state/actions/userActions";
+import {
+    getMajors,
+    getPlans,
+    getMajorFromState,
+    getUserCoopCycleFromState,
+    getFullNameFromState,
+    getToken,
+    getId,
+    getEmail
+  } from "../state";
+import { Dispatch } from "redux";
+import { Major, Schedule } from "graduate-common";
+import { AppState } from "../state/reducers/state";
+import { planToString } from "../utils";
+import { updateUser } from "../services/UserService"
+import { IUpdateUser, IUpdateUserData } from "../models/types";
+
+
+const OuterContainer = styled.div`
+    width: 50%;
+    margin: 0 auto;
+`;
+
+const InnerContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+    border: 3px solid #EB5757;
+    box-sizing: border-box;
+    margin: 10%;
+    padding: 30px 20% 40px 30px;
+`;
+
+const ProfileTitle = styled.h1`
+    padding-right: 10px;
+`;
+
+const ProfileTitleContainer = styled.div`
+    display: flex;
+    flex-direction: row;
+`;
+
+const DataContainer = styled.div`
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+`;
+
+const ProfileColumn = styled.div`
+    display: flex;
+    flex-direction: column;
+`;
+
+const ProfileEntryContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+    padding: 15px 0px 15px 0px;
+`;
+
+const ItemTitle = styled.p`
+    font-weight: bold;
+    margin: 0;
+    padding: 5px 0px 5px 0px;
+`;
+
+const ItemEntry = styled.p`
+    margin: 0;
+    padding: 0;
+`;
+
+const Container = styled.div`
+`;
+
+const HeaderRow = styled.div`
+    display: flex;
+    justify-content: space-between;
+    padding: 0px 20px 0px 20px;
+`;
+
+const StyledLink = styled(Link)`
+  padding: 20px 0px 0px 0px;
+  color: #EB5757;
+`
+interface ProfileNameProps {
+    name: string,
+    isEdit: boolean,
+    setName: (name: string) => void
+}
+
+interface ProfileMajorProps {
+    major: string,
+    isEdit: boolean,
+    majors: Major[],
+    setMajor: (major: Major | undefined) => void
+}
+
+interface ProfileCoopProps {
+    major: Major,
+    coop: string,
+    isEdit: boolean,
+    plans: Record<string, Schedule[]>,
+    setCoop: (coop: Schedule | undefined) => void
+}
+
+interface ProfileEmailProps {
+    email: string,
+    isEdit: boolean,
+    setEmail: (email: string) => void
+}
+
+interface SaveProps {
+    setEdit: Function,
+    name: string,
+    email: string,
+    major: Major,
+    coop: string,
+    token: string,
+    id: number,
+    setMajorAction: (major?: Major) => void,
+    setCoopCycleAction: (plan: string) => void,
```

Comment:
The  setCoopCycleAction actually takes in a schedule (you can refactor it to take in a string, or just convert it in save)

Reply:
Same as above